R/capture-requests.R

```diff
@@ -222,6 +230,10 @@ dbSendQueryTrace <- quote({
 })
 
 dbListTablesTrace <- quote({
+  if (is.null(.dittodb_env$db_path)) {
+    # Inherits the error call from dbSendQueryTrace
+    return(invisible())
+  }
```

Comment:
What about, instead of each of these `if`s and `return(invisible())`s you made a helper function that was something like (:

```
ensure_dittodb_env_setup <- function() {
 if (is.null(.dittodb_env$db_path)) {
    rlang::abort(...)
  }
}
```

And then called `ensure_dittodb_env_setup()` in each of these directly?

Reply:
You'll get a kick out of this. Adding an abort there triggers the error from dbSendQuery

![image](https://github.com/ropensci/dittodb/assets/23486577/c42142f9-fef6-4a5b-96d3-de674d0b934d)

Results in

![image](https://github.com/ropensci/dittodb/assets/23486577/dc820946-0f05-4fff-9d2d-6a4516f50256)

I was trying to figure out how to do a silent abort in a function like you mentioned, I will put some more research into it to see if I can come up with anything

Reply:
okay, figure it out :D i had to be a bit sneaky to avoid the duplicate messages. i also removed some of the checks from functions where i do not think the belong.